src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.cpp

```diff
@@ -0,0 +1,501 @@
+{
+    // According to the Matter WebRTC Transport Provider specification,
+    // at least one of VideoStreamID or AudioStreamID must be present.
+    if (!req.videoStreamID.HasValue() && !req.audioStreamID.HasValue())
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::ConstraintError);
+        return;
+    }
+
+    // Prepare arguments for Delegate::HandleSolicitOffer using OfferRequestArgs struct
+    Delegate::OfferRequestArgs args;
+    args.id                 = GenerateSessionID();
+    args.streamUsage        = req.streamUsage;
+    args.videoStreamID      = req.videoStreamID;
+    args.audioStreamID      = req.audioStreamID;
+    args.iceServers         = req.ICEServers;
+    args.iceTransportPolicy = req.ICETransportPolicy;
+    args.metadataOptions    = req.metadataOptions;
+    args.peerNodeID         = GetNodeIdFromCtx(ctx.mCommandHandler);
+    args.peerFabricIndex    = ctx.mCommandHandler.GetAccessingFabricIndex();
+
+    // Delegate processing
+    WebRTCSessionTypeStruct outSession;
+    bool deferredOffer = false;
+
+    auto status = Protocols::InteractionModel::ClusterStatusCode(mDelegate->HandleSolicitOffer(args, outSession, deferredOffer));
+    if (!status.IsSuccess())
+    {
+        // Delegate encountered a problem (out of resources, invalid data, etc.)
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+        return;
+    }
+
+    // Store or update the WebRTCSessionTypeStruct in the CurrentSessions
+    if (UpsertSession(outSession) == UpsertResultEnum::kInserted)
+    {
+        ChipLogProgress(Zcl, "WebRTCTransportProvider: Inserted a new session with ID=%u.", outSession.id);
+    }
+    else
+    {
+        ChipLogProgress(Zcl, "WebRTCTransportProvider: Updated existing session with ID=%u.", outSession.id);
+    }
+
+    // Send the response
+    Commands::SolicitOfferResponse::Type resp;
+    resp.webRTCSessionID = outSession.id;
+    resp.deferredOffer   = deferredOffer;
+    resp.videoStreamID.SetValue(outSession.videoStreamID);
+    resp.audioStreamID.SetValue(outSession.audioStreamID);
+
+    ctx.mCommandHandler.AddResponse(ctx.mRequestPath, resp);
+}
+
+void WebRTCTransportProviderServer::HandleProvideOffer(HandlerContext & ctx, const Commands::ProvideOffer::DecodableType & req)
+{
+    // Extract command fields from the request.
+    auto webRTCSessionID = req.webRTCSessionID;
+    auto videoStreamID   = req.videoStreamID;
+    auto audioStreamID   = req.audioStreamID;
+
+    NodeId peerNodeId           = GetNodeIdFromCtx(ctx.mCommandHandler);
+    FabricIndex peerFabricIndex = ctx.mCommandHandler.GetAccessingFabricIndex();
+
+    uint16_t sessionId = 0;
+
+    // Determine if this is a new session or a re-Offer
+    if (webRTCSessionID.IsNull())
+    {
+        // This is a request for a new session
+        // If both the VideoStreamID and AudioStreamID are not present
+        if (!videoStreamID.HasValue() && !audioStreamID.HasValue())
```

Comment:
This check does not depend on whether the WebRTCSessionID is null, in the spec.  Needs to be done unconditionally.

Reply:
"If the accessing Peer Node ID and Local Fabric Index entries stored in the Secure Session Context, does not equal the PeerNodeID and FabricIndex for the WebRTCSessionID entry in [CurrentSessions]"

Note: FabricIndex has been removed from WebRTCSessionStruct

Reply:
That is for SolicitOffer, for ProvideOffer, there is "If WebRTCSessionID is null:"